scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,618 @@
+#
+###############################
+#
+#    ^ notice "
+#    |
+#<---Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users independently.
+#    "
+#    fi
+#
+###############################
+#
+# This produces output in color:
+#
+###############################
+#
+#./install: Notice:
+#
+#    Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users independently.
+#
+###############################
+#
+true # so {...} is not empty, shell will not permit otherwise
+}
+###############################
+###  Setup environment
+###############################
+{
+# Set the character collating sequence to be numeric ASCII/C standard.
+# Set the character set to be standard one-byte ASCII.
+# Set default umask; to be non-restrictive and friendly to others.
+#
+LC_COLLATE=C
+LANG=C
+umask 022
+}
+###############################
+###  Main constants
+###############################
+{
+readonly dest="/nix"
+readonly self="$(dirname "$0")"
+readonly nix="@nix@"
+readonly cacert="@cacert@"
+readonly appname="$0"
+}
+###############################
+###  CLI control constants
+###############################
+{
+# tput is not present on some systems (Alpine Linux), this trick with printf allows to store, not 'codes' - literal symbols
+readonly red=$(printf '\033[1;31m')
+readonly green=$(printf '\033[1;32m')
+readonly yellow=$(printf '\033[1;33m')
+readonly blue=$(printf '\033[1;34m')
+readonly bold=$(printf '\033[1m')
+readonly reset=$(printf '\033[0;m') # Reset to default output
+}
+###############################
+###  CLI output functions
+###############################
+{
+print() {
+    # Unified output function
+    # Using printf, because it is more portable across all platforms than echo, and it is important for Nix install
+    #
+    if [ -z "$message" ]; then
+        message="$1"
+    fi
+    if [ -z "$color" ]; then
+        color="$reset"
+    fi
+    if [ -z "$prefix" ]; then
+        prefix='Info'
+    fi
+    # This line makes all prints in script
+    # Form:
+    ### Application: Prefix: Body of message
+    printf '%s%s: %s: %s%s\n' "$color" "$appname" "$prefix" "$message" "$reset"
+
+    # Output is done, unset variables that could've been set by parent functions
+    unset message
+    unset color
+    unset prefix
+}
+
+notice() {
+    message="$1"
+    color="$green"
+    prefix='Notice'
+    print
+}
+
+warning() {
+    message="$1"
+    color="$yellow"
+    prefix='Warning'
+    >&2 print
+}
+
+error() {
+    message="$1"
+    exitSign="$2"
+    color="$red"
+    prefix='Error'
+    >&2 print
+    if [ -z "$exitSign" ]; then
+        exit 1
+    fi
+}
+
+errorRevert() {
+    message="$1"
+    exitSign="$2"
```

Comment:
Ok.

exitSign is really not used there. Since that function is not for exit.

But I am thinking how to with taste combine `error` and `errorRevert` functions.